types/koa-cash/index.d.ts

```diff
@@ -0,0 +1,83 @@
+// Type definitions for koa-cash 4.0
+// Project: https://github.com/koajs/cash
+// Definitions by: Jakob Goebel <https://github.com/jagoe>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import * as Koa from 'koa';
+declare module 'koa' {
+    interface BaseContext {
+        /**
+         * This is how you enable a route to be cached. If you don't call await ctx.cashed(),
+         * then this route will not be cached nor will it attempt to serve the request from the cache.
+         *
+         * Notes:
+         * * Only `GET` and `HEAD` requests are cached.
+         * * Only 200 responses are cached. Don't set 304 status codes on these routes - this
+         *   middleware will handle it for you.
+         * * The underlying store should be able to handle Date objects as well as Buffer objects.
+         *   Otherwise, you may have to serialize/deserialize yourself.
+         * @param maxAge The max age passed to `get()`.
+         */
+        cashed(maxAge?: number): Promise<boolean>;
+    }
+}
+
+declare function koaCash(opts?: cash.Options): Koa.Middleware;
+
+declare namespace cash {
+    interface Options {
```

Comment:
This interface won't be reachable externally - is that what you want? Or would you rather merge this namespace with the exported `koaCash` function (by naming it the same thing) and thereby exposing this interface?

Reply:
Good point, thank you. The `koaCash.Options` should now get exposed as well.